src/app/containers/ArticleTimestamp/helpers.test.js

```diff
@@ -1,20 +1,137 @@
+    });
+  });
+
+  describe('isLastRelative', () => {
+    const timeLimits = generateTimeLimits(15);
+    timeLimits.forEach(({ expectedValue, time, title }) => {
+      it(`should return ${expectedValue} when lastPublished is ${title}`, () => {
+        const lastPublished = time;
+        expect(isLastRelative(lastPublished)).toBe(expectedValue);
+      });
+    });
+  });
+
+  describe('isSameDay', () => {
+    it('should return true if both timestamps are on the same day', () => {
+      const firstPublished = 1569962265000;
+      const lastPublished = 1569962145008;
+      expect(isSameDay(firstPublished, lastPublished)).toBe(true);
+    });
+
+    it('should return false if both timestamps are not on the same day', () => {
+      const fourHoursAgo = timestampGenerator({ hours: 4 });
+      const thirtySixHoursAgo = timestampGenerator({ hours: 36 });
+      expect(isSameDay(fourHoursAgo, thirtySixHoursAgo)).toBe(false);
+    });
+  });
+
+  describe('isToday', () => {
+    it('should return true if timestamp is today (3 minutes ago)', () => {
+      const timestamp = timestampGenerator({ minutes: 3 });
+      expect(isToday(timestamp)).toBe(true);
+    });
+
+    it('should return false if timestamp is not today (32 hours ago)', () => {
+      const timestamp = timestampGenerator({ hours: 32 });
+      expect(isToday(timestamp)).toBe(false);
+    });
+  });
+
+  describe('formatType', () => {
+    const dateFormats = {
+      date: 'D MMMM YYYY',
+      dateTimeTimezone: 'D MMMM YYYY, HH:mm z',
+    };
+    it(`should return default date format when firstPublished is not today`, () => {
+      const firstPublished = timestampGenerator({ days: 5 });
+      expect(formatType({ firstPublished })).toBe(dateFormats.date);
+    });
+
+    it(`should return dateTimeTimezone format when firstPublished is today`, () => {
+      const firstPublished = timestampGenerator({ hours: 4 });
+      expect(formatType({ firstPublished })).toBe(dateFormats.dateTimeTimezone);
+    });
+
+    it(`should return date format when lastPublished and firstPublished are not on the same day`, () => {
+      const firstPublished = timestampGenerator({ hours: 50 });
+      const lastPublished = timestampGenerator({ hours: 20 });
+      expect(formatType({ firstPublished, lastPublished })).toBe(
+        dateFormats.date,
+      );
+    });
+
+    it(`should return dateTimeTimezone format when firstPublished and lastPublished are on the same day and today`, () => {
+      const firstPublished = timestampGenerator({ hours: 4 });
+      const lastPublished = timestampGenerator({ hours: 2 });
+      expect(formatType({ firstPublished, lastPublished })).toBe(
+        dateFormats.dateTimeTimezone,
+      );
+    });
+
+    it(`should return default date format when firstPublished is null`, () => {
+      const firstPublished = null;
+      expect(formatType({ firstPublished })).toBe(dateFormats.date);
```

Comment:
I don't think this test correctly reflects the logic in the implementation due to the way default parameters work. Here we are passing an explicit `null` to `formatType` as the first argument (lastPublished), not setting firstPublished to null. And no I'm not sure why formatDate was implemented that way :-/

Reply:
@jamesdonoh `lastPublished` isn't really the first argument as `formatType` takes in an object, `firstPublished` will be null and `lastPublished` will default to `null`

Reply:
You're absolutely right, apologies. I missed the curly brackets :-)